compiler/include/baseAST.h

```diff
@@ -543,6 +547,17 @@ static inline const CallExpr* toConstCallExpr(const BaseAST* a)
   case E_TryStmt:                                                       \
     AST_CALL_CHILD(_a, TryStmt, body(), call, __VA_ARGS__);             \
     break;                                                              \
+  case E_ForallStmt: {                                                  \
+    AST_CALL_LIST (_a, ForallStmt,    iterExprs(),  call, __VA_ARGS__); \
+    ForallIntents* fi = ((ForallStmt*)_a)->withClause();                \
+    AST_CALL_STDVEC(fi->fiVars,  Expr,              call, __VA_ARGS__); \
+    AST_CALL_STDVEC(fi->riSpecs, Expr,              call, __VA_ARGS__); \
+    AST_CALL_CHILD(fi, ForallIntents, iterRec,      call, __VA_ARGS__); \
+    AST_CALL_CHILD(fi, ForallIntents, leadIdx,      call, __VA_ARGS__); \
+    AST_CALL_CHILD(fi, ForallIntents, leadIdxCopy,  call, __VA_ARGS__); \
+    AST_CALL_CHILD(_a, ForallStmt,    masterBlock(),call, __VA_ARGS__); \
```

Comment:
Why can't this be something like
```
AST_CALL_CHILD(_a, BlockStmt, fMBlock, call, __VA_ARGS__);
AST_CALL_LIST(_a, ?, fIter, call, __VA_ARGS__);
AST_CALL_CHILD(_a, ForallIntents, fWith, call, __VA_ARGS__);
AST_CALL_CHILD(fWith);
```

?

I'm worried there will be bugs if everything in the pointed-to AST node is not visited here...